modules/lvgl/input/lvgl_button_input.c

```diff
@@ -45,13 +50,30 @@ static void lvgl_button_process_event(const struct device *dev, struct input_eve
 	}
 }
 
+static void lvgl_button_read_cb(lv_indev_drv_t *drv, lv_indev_data_t *data)
+{
+	const struct device *dev = drv->user_data;
+	const struct lvgl_common_input_config *cfg = dev->config;
+	struct lvgl_button_input_data *button_data = dev->data;
+
+	if (k_msgq_get(cfg->event_msgq, data, K_NO_WAIT) != 0) {
+		*data = button_data->previous_state;
+		goto release;
+	}
+
+	button_data->previous_state = *data;
```

Comment:
Normally I'd expect a `memcpy` for copying a struct data... though I guess assignment works but it feels out of place. @gmarull are you aware of any drawback of this? Don't think it's covered by either our or Linux style guide but I may be missing something.

Reply:
Yeah while writing this I took inspiration from the kscan driver one. I also had a version with memcpy at some point, my suspicion is that the compiler will optimise this to a memcpy anyway. But since we most likely can/dont want to make assumption about the compilers optimizers I'll make it explicit :^)

Reply:
sounds good